lib/class-wp-rest-plugins-controller.php

```diff
@@ -0,0 +1,822 @@
+					'args'                => $this->get_collection_params(),
+				),
+				array(
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'create_item' ),
+					'permission_callback' => array( $this, 'create_item_permissions_check' ),
+					'args'                => array(
+						'slug'   => array(
+							'type'        => 'string',
+							'required'    => true,
+							'description' => __( 'WordPress.org plugin directory slug.', 'gutenberg' ),
+						),
+						'status' => array(
+							'description' => __( 'The plugin activation status.', 'gutenberg' ),
+							'type'        => 'string',
+							'enum'        => is_multisite() ? array( 'inactive', 'active', 'network-active' ) : array( 'inactive', 'active' ),
+							'default'     => 'inactive',
+						),
+					),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<plugin>' . self::PATTERN . ')',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => array( $this, 'get_item_permissions_check' ),
+				),
+				array(
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => array( $this, 'update_item' ),
+					'permission_callback' => array( $this, 'update_item_permissions_check' ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::EDITABLE ),
+				),
+				array(
+					'methods'             => WP_REST_Server::DELETABLE,
+					'callback'            => array( $this, 'delete_item' ),
+					'permission_callback' => array( $this, 'delete_item_permissions_check' ),
+				),
+				'args'   => array(
+					'context' => $this->get_context_param( array( 'default' => 'view' ) ),
+					'plugin'  => array(
+						'type'              => 'string',
+						'pattern'           => self::PATTERN,
+						'validate_callback' => static function ( $file ) {
+							if ( ! is_string( $file ) || ! preg_match( '/' . self::PATTERN . '/u', $file ) ) {
+								return false;
+							}
+
+							$validated = validate_file( plugin_basename( $file ) );
+
+							return 0 === $validated;
+						},
+						'sanitize_callback' => static function ( $file ) {
+							return plugin_basename( sanitize_text_field( $file . '.php' ) );
+						},
+					),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Checks if a given request has access to get plugins.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has read access, WP_Error object otherwise.
+	 */
+	public function get_items_permissions_check( $request ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		if ( ! current_user_can( 'activate_plugins' ) ) {
```

Comment:
What about the `manage_network_plugins`? 
https://github.com/WordPress/wordpress-develop/blob/master/src/wp-admin/network/menu.php#L105

Reply:
That is handled by `map_meta_cap`, https://github.com/WordPress/wordpress-develop/blob/cc62b4ba1da5dc319e28f5538ea4936451bd3c7c/src/wp-includes/capabilities.php#L476.

See also `check_read_permission` which excludes network only plugins from being in the plugins list for users who can't manage network plugins if the plugin is not active.